src/pages/api/graphql.ts

```diff
@@ -130,25 +147,48 @@ interface TransferAssetVars {
 }
 
 const mutationResolvers: MutationResolvers<AppContext> = {
-  async mint(_a, _b, { session, dataSources: { db, holaplex } }) {
+  async mint(_a, _b, { session, dataSources: { db, holaplex, user } }) {
     if (!session) {
       return null;
     }
 
-    const wallet = await db.wallet.findFirst({
-      where: {
-        user: {
-          email: session?.user?.email
-        }
+    const dropResponse = await holaplex.query<GetDropData, GetDropVars>({
+      fetchPolicy: 'network-only',
+      query: GetProjectDrop,
+      variables: {
+        project: process.env.HOLAPLEX_PROJECT_ID as string,
+        drop: process.env.HOLAPLEX_DROP_ID as string
       }
     });
 
+    const blockchain = dropResponse.data.project.drop?.collection.blockchain;
+
+    const me = await user.get(session.user?.email);
+    const wallets = me?.wallets;
+
+    let assetId = '';
+    switch (blockchain) {
+      case Blockchain.Solana:
+        assetId = AssetType.Sol;
+        break;
+      case Blockchain.Polygon:
+        assetId = AssetType.Matic;
+        break;
+      case Blockchain.Ethereum:
+        assetId = AssetType.Eth;
+        break;
+    }
+
+    const recipient = wallets?.filter(
+      (wallet) => wallet?.assetId === assetId
+    )[0]?.address;
```

Comment:
nice this is good. instead of calling me.wallets take the user.customerId and then have a query to fetch customer {wallet(assetId: $assetId)} based on the blockchain of the drop.